waiter/src/waiter/util/ring_utils.clj

```diff
@@ -42,6 +43,29 @@
   (let [encoding (or (ring-request/character-encoding request) "UTF-8")]
     (ring-params/assoc-query-params request encoding)))
 
+(defn conj-value
```

Comment:
We can remove `conj-value` and `request->-query-params` and instead use `query-params-request` if we use URL-safe Base64 encoding.